job_controller/job_controller.go

```diff
@@ -320,3 +369,201 @@ func (jc *JobController) resolveControllerRef(namespace string, controllerRef *m
 	}
 	return job
 }
+
+// reconcileJobs checks and updates replicas for each given ReplicaSpec.
+// It will requeue the job in case of an error while creating/deleting pods/services.
+func (jc *JobController) reconcileJobs(job interface{}, replicas map[commonv1.ReplicaType]*commonv1.ReplicaSpec,
```

Comment:
ReconcileJobs is misleading in JobController

Reply:
Do you mean the name is misleading ? 
This is the entrypoint that is supposed to be called by TFJob etc. reconcile loop.

Reply:
I meant, Reconcile function looks very general on a whole. When we move into the common code, its interface should be well defined. Else, new developers might not be able to use it.

Reply:
Actually, reconcileJobs currently is not an interface. it provides concrete implementations.
The custom operators has to somehow invoke (or be invoked by) the library, and that is this entrypoint method.
We should write proper documentations and examples for how to use it.
Do you have any suggestions for how to construct this?